src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3069,6 +3069,11 @@ protected override void VisitArguments(ImmutableArray<BoundExpression> arguments
                 }
             }
 
+            if (method is object && (method.FlowAnalysisAnnotations & FlowAnalysisAnnotations.DoesNotReturn) == FlowAnalysisAnnotations.DoesNotReturn)
```

Comment:
The LDM notes seem to indicate that DoesNotReturn makes the code below *unreachable*, not unreachable for nullable analysis.

Reply:
Filed https://github.com/dotnet/roslyn/issues/37081